simple-page-ordering.php

```diff
@@ -219,6 +223,38 @@ public static function ajax_simple_page_ordering() {
 			die( wp_json_encode( $result ) );
 		}
 
+		/**
+		 * Page ordering reset ajax callback
+		 *
+		 * @return void
+		 */
+		public static function ajax_reset_simple_page_ordering() {
+			global $wpdb;
+
+			$nonce = isset( $_POST['_wpnonce'] ) ? sanitize_key( wp_unslash( $_POST['_wpnonce'] ) ) : '';
+
+			if ( ! wp_verify_nonce( $nonce, 'simple-page-ordering-nonce' ) ) {
+				die( -1 );
+			}
+
+			// check and make sure we have what we need
+			$post_type = isset( $_POST['post_type'] ) ? sanitize_text_field( wp_unslash( $_POST['post_type'] ) ) : '';
+
+			if ( empty( $post_type ) ) {
+				die( -1 );
+			}
+
+			// does user have the right to manage these post objects?
+			if ( ! self::check_edit_others_caps( $post_type ) ) {
+				die( -1 );
+			}
+
+			// reset the order of all posts of given post type
+			$wpdb->update( 'wp_posts', array( 'menu_order' => 0 ), array( 'post_type' => $post_type ) );
```

Comment:
@dkotter This PR works well. The only area that I am concerned about is the performance impact due to this line.
Would this be an issue for an extremely large database? If so, should we look into performing the updates in batches?

Reply:
Looking at the code [`wpdb::update`](https://developer.wordpress.org/reference/classes/wpdb/update/) runs, seems like the final query will be something like:
```sql
UPDATE 'wp_posts' SET 'menu_order' = 0 WHERE 'post_type' = 'page'
``` 

I think that should scale fairly well and we won't need to worry about doing any sort of batching here. That said, @ruscoe if you'd be able to do some quick testing on various sizes to see the performance impact, that would help us be more confident about merging this in. Something like profiling this code when run on a site that has 10 pages, 100 pages and 1000 pages, that would be helpful.

In addition, I think best practice here is to utilize the format properties of the `update` method. Something like
```suggestion
			$wpdb->update( 'wp_posts', array( 'menu_order' => 0 ), array( 'post_type' => $post_type ), array( '%d' ), array( '%s' ) );
```

Reply:
@dkotter No problem at all! I profiled by measuring execution time.

Just to show my process, first I used a quick shell script to create pages:

```bash
#!/bin/bash

for i in {1..10}
do
  wp post create --post_type=page --post_title="Page $i"
done
```

Then used the following to test:

```php
$start_time = microtime(true);
$wpdb->update( 'wp_posts', array( 'menu_order' => 0 ), array( 'post_type' => 'page' ), array( '%d' ), array( '%s' ) );
$end_time = microtime(true);
$total_time = ( $end_time - $start_time );
echo $total_time;
```

Taking the average of five test runs each:

10 posts: 0.0041 microseconds

100 posts: 0.0042 microseconds

1000 posts: 0.0116 microseconds

5000 posts: 0.0299 microseconds

How does that look?

Reply:
Thanks @ruscoe! Definitely see an increase in time as the number of posts increase but the overall amount of time is so low that this doesn't seem like it will cause any problems. This looks good to go on my end